gluten-core/src/main/scala/org/apache/spark/sql/execution/GlutenFallbackReporter.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution
+
+import io.glutenproject.GlutenConfig
+import io.glutenproject.extension.GlutenPlan
+import io.glutenproject.extension.columnar.{TRANSFORM_UNSUPPORTED, TransformHints}
+import io.glutenproject.utils.LogLevelUtil
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.rules.Rule
+
+/**
+ * This rule is used to collect all fallback reason.
+ * 1. print fallback reason for each plan node
+ * 2. TODO: post all fallback reason using one event
+ */
+case class GlutenFallbackReporter(
+    glutenConfig: GlutenConfig,
+    spark: SparkSession) extends Rule[SparkPlan] with LogLevelUtil {
+
+  override def apply(plan: SparkPlan): SparkPlan = {
+    if (!glutenConfig.enableFallbackReport) {
+      return plan
+    }
+    printFallbackReason(plan)
```

Comment:
can this ` printFallbackReason(plan)` be put ahead of check `enableFallbackReport`?  We'd allow fallback reasons dump into log even not show in UI.

Reply:
I'm fine with this

Reply:
or, we can add one more config

Reply:
No need to add one more config. Change `logOnLevel` to `logInfo` and do a check for `printFallbackReason` by `validateFailureLogLevel == DEBUG` and keep default of `validateFailureLogLevel` to DEBUG. It should work as previous print logic.

Reply:
I wonder why we need do `validateFailureLogLevel == DEBUG`, why not `logOnLevel(validateFailureLogLevel)` ?

Reply:
We want not to print fallback log when validateFailureLogLevel is `INFO` but by default we print the logs so we set it to `validateFailureLogLevel ` to `DEBUG` by default. The logic is a little tricky so I am okay if you keep using `logOnLevel`. Just make sure by default it's dumped.